pkg/controller/volume/pvcprotection/pvc_protection_controller_test.go

```diff
@@ -400,6 +443,142 @@ func testPVCProtectionController(t *testing.T, genericEphemeralVolumeFeatureEnab
+			initialObjects: []runtime.Object{
+				withVolInUse(node(), otherVolumeName),
+			},
+			informersAreLate: true,
+			// PVC does not contain selected-node annotation.
+			updatedPVC: deleted(withSelectedNodeAnnotation(withProtectionFinalizer(pvcWithVolumeName(pvc())))),
+			expectedActions: []clienttesting.Action{
+				// Pods listed, no pod found referencing the pvc
+				clienttesting.NewListAction(podGVR, podGVK, defaultNS, metav1.ListOptions{}),
+				// node informer is empty, hence nodes listed from API server
+				clienttesting.NewListAction(nodeGVR, nodeGVK, "", metav1.ListOptions{}),
+				//PVC finalizer removed.
+				clienttesting.NewUpdateAction(pvcGVR, defaultNS, deleted(withSelectedNodeAnnotation(pvcWithVolumeName(pvc())))),
+			},
+			storageObjectInUseProtectionEnabled: true,
+		},
+		{
+			name: "StorageObjectInUseProtection Enabled, deleted PVC with a volume name, finalizer and node selector set, node informer contains node, different volume found in the node.status.volumesInUse, API server node list invoked -> finalizer is removed",
+			initialObjects: []runtime.Object{
+				withVolInUse(node(), otherVolumeName),
+			},
+			// PVC does not contain selected-node annotation.
+			updatedPVC: deleted(withSelectedNodeAnnotation(withProtectionFinalizer(pvcWithVolumeName(pvc())))),
+			expectedActions: []clienttesting.Action{
+				// Pods listed, no pod found referencing the pvc
+				clienttesting.NewListAction(podGVR, podGVK, defaultNS, metav1.ListOptions{}),
+				// node informer is did not report a node with the given volume, hence nodes listed from API server
+				clienttesting.NewListAction(nodeGVR, nodeGVK, "", metav1.ListOptions{}),
+				//PVC finalizer removed.
+				clienttesting.NewUpdateAction(pvcGVR, defaultNS, deleted(withSelectedNodeAnnotation(pvcWithVolumeName(pvc())))),
+			},
+			storageObjectInUseProtectionEnabled: true,
+		},
+		{
+			name: "StorageObjectInUseProtection Enabled, deleted PVC with a volume name, finalizer, node informer contains node, different volume found in the node.status.volumesInUse, API server node list invoked -> finalizer is removed",
+			initialObjects: []runtime.Object{
+				withVolInUse(node(), otherVolumeName),
+			},
+			// PVC does not contain selected-node annotation.
+			updatedPVC: deleted(withProtectionFinalizer(pvcWithVolumeName(pvc()))),
+			expectedActions: []clienttesting.Action{
+				// Pods listed, no pod found referencing the pvc
+				clienttesting.NewListAction(podGVR, podGVK, defaultNS, metav1.ListOptions{}),
+				// node informer is did not report a node with the given volume, hence nodes listed from API server
+				clienttesting.NewListAction(nodeGVR, nodeGVK, "", metav1.ListOptions{}),
+				//PVC finalizer removed.
+				clienttesting.NewUpdateAction(pvcGVR, defaultNS, deleted(pvcWithVolumeName(pvc()))),
+			},
+			storageObjectInUseProtectionEnabled: true,
+		},
```

Comment:
Do you have a test case where a node has the volume marked in use but the PVC isn't stamped with the volume name?

It shouldn't matter given the current code but might be useful to check in order to prevent regressions.

Reply:
to understand correctly, PVC spec.VolumeName not set, and node.status.VolumesInUse = ['vol1'], not sure how this situation can arise. However I have added a test case, we skip the check for volumes in use since pvc spec volume name is missing.

Reply:
Yeah the point is to check the code path in case something changes in the future.